### PR TITLE
Remove libcugraphops dependency and add CUDA 12 CI support

### PR DESCRIPTION
**Pull request by dongxuy04**
As `libcugraphops` is only needed to run GNN applications, not required by `libwholgraph` or `pylibwholegraph`, just remove it from dependency.
Updated YAML files to support CUDA 12 CI.
Fix exit code in `ci/test_python.sh` and also skip python test for CUDA 12 as no public stable pytorch package.

### CODE REVIEW

File: conda/recipes/libwholegraph/meta.yaml [hunk 1]
```diff
@@ -41,19 +41,28 @@ build:
...
+    - {{ compiler('cuda11') }} ={{ cuda_version }}
+    {% else %}
+    - {{ compiler('cuda') }}
+    {% endif %}
+    - cuda-version ={{ cuda_version }}
+    {% if cuda_major == "11" %}
+    - cudatoolkit
+    {% else %}
+    - cuda-cudart-dev
+    - cuda-driver-dev
```
Comment (tingyu66) [inline conda/recipes/libwholegraph/meta.yaml:55]: I wonder why wholegraph needs CUDA driver library.

**Review by tingyu66 — APPROVED** [review]
Just have a suggestion and a question. LGTM, otherwise.

File: conda/recipes/libwholegraph/meta.yaml (hunk 1 above)
Comment (tingyu66) [reply conda/recipes/libwholegraph/meta.yaml:55]: Nevermind, just found out the source code that uses memory management functions from CUDA driver API.

File: dependencies.yaml [hunk 2]
```diff
@@ -122,6 +127,11 @@ dependencies:
             packages:
               - cudatoolkit=11.8
               - cuda-nvtx=11.8
+          - matrix:
+              cuda: "12.0"
+            packages:
+              - cuda-version=12.0
```
[the remark this answers is not in the document]
Comment (dongxuy04) [reply dependencies.yaml:133]: Thanks and commited!

**Comment by tingyu66** [comment]
@BradReesWork Shall we merge this PR into refactoring?

**Review by BradReesWork — APPROVED** [review]

**Pull request merged**